### PR TITLE
Implement list() method counterfactual manager

### PR DESCRIPTION
**Pull request by gaugup**

### CODE REVIEW

File: responsibleai/responsibleai/_managers/counterfactual_manager.py [hunk 1]
```diff
@@ -575,7 +580,23 @@ def _get_counterfactual_schema(version):
             return json.load(f)
 
     def list(self):
-        pass
+        """List information about the CounterfactualManager.
```
Comment (imatiach-msft) [inline responsibleai/responsibleai/_managers/counterfactual_manager.py:583]: wow, how has this list method not been implemented yet :sweat_smile: is there anything else we are still missing?

**Review by imatiach-msft — APPROVED** [review]

File: responsibleai/responsibleai/_managers/counterfactual_manager.py [hunk 2]
```diff
@@ -206,20 +207,24 @@ def load_config(config_directory_path):
...
         """
         result = {}
-        if self.counterfactual_obj is not None:
-            result[CounterfactualConfig.COUNTERFACTUAL_OBJ] = \
-                self.counterfactual_obj.to_json()
-        else:
-            result[CounterfactualConfig.COUNTERFACTUAL_OBJ] = None
+        if include_counterfactual_obj:
+            if self.counterfactual_obj is not None:
+                result[CounterfactualConfig.COUNTERFACTUAL_OBJ] = \
```
Comment (imatiach-msft) [inline responsibleai/responsibleai/_managers/counterfactual_manager.py:224]: minor nit: I think you can avoid several slashes by making CounterfactualConfig.COUNTERFACTUAL_OBJ a constant at the top, as well as some other constants referenced here.
Something like:
```
COUNTERFACTUAL_OBJ = CounterfactualConfig.COUNTERFACTUAL_OBJ
HAS_COMPUTATION_FAILED = CounterfactualConfig.HAS_COMPUTATION_FAILED
...
```

File: responsibleai/responsibleai/_managers/counterfactual_manager.py (hunk 1 above)
Comment (gaugup) [reply responsibleai/responsibleai/_managers/counterfactual_manager.py:583]: Hopefully this is the last one.

File: responsibleai/responsibleai/_managers/counterfactual_manager.py (hunk 2 above)
Comment (gaugup) [reply responsibleai/responsibleai/_managers/counterfactual_manager.py:224]: will probably keep it thi way.

**Pull request merged**